File: leetcode/0003/lengthOfLongestSubstring.py
def lengthOfLongestSubstring(s: str) -> int:
    if not s: return 0
    if len(s) == 0: return 0
    resList = []
    ans = 0
    for i in range(len(s)):
        if s[i] in resList:
            index = resList.index(s[i])
            resList = resList[index+1:]
            resList.append(s[i])
        else:
            resList.append(s[i])
            if ans < len(resList):
                ans = len(resList)
    return ans

res = lengthOfLongestSubstring('abcabcbb')
print(res)

File: leetcode/0003/lengthOfLongestSubstring.swift
func lengthOfLongestSubstring(_ s: String) -> Int {
    guard s.count > 0 else {
        return 0
    }
    var list: Array<Character> = []
    var ans = 0
    for char in s {
        if let firstIndex = list.firstIndex(of: char) {
            list = Array(list[firstIndex+1..<list.endIndex])
            list.append(char)
        } else {
            list.append(char)
            ans = ans < list.count ? list.count : ans
        }
        print(list)
    }
    return ans
}

let res = lengthOfLongestSubstring("pwwkew")
print(res)
